lib/rubocop/cop/lint/out_of_range_regexp_ref.rb

```diff
@@ -38,11 +41,23 @@ def on_send(node)
 
           if node.receiver&.regexp_type?
             check_regexp(node.receiver)
-          elsif node.first_argument&.regexp_type? && node.method?(:=~)
+          elsif node.first_argument&.regexp_type? \
+            && REGEXP_ARGUMENT_METHODS.include?(node.method_name)
             check_regexp(node.first_argument)
           end
         end
 
+        def on_when(node)
+          @valid_ref = nil
+
+          regexp_conditions = node.conditions.select(&:regexp_type?)
+          return if regexp_conditions.empty?
```

Comment:
You could remove this and `@valid_ref = nil`, as `[].max # => nil`. Might not be as clear though, your choice

Reply:
This was a case of red, green, no refactor. :) I'll ✂️